Readme.md

```diff
@@ -3,29 +3,59 @@
-- Set the <strong>FileUploadCompleteEventArgs.CallbackData </strong>property to "Images/<filename>", where <filename> is a full name of the file you saved. Please note that you need to save the file to directory "~/Images/", where "~/" is the root folder of your project, but set the <strong>FileUploadCompleteEventArgs.CallbackData </strong>property to "Images/<filname>" <u>excluding</u> the root folder.<br />
-- In the client-side event handler, use the <strong>document.getElementById </strong>method to find your <strong>html image</strong> by the assigned <strong>id</strong>.<br />
-- Set the <strong>src </strong>property of the found image to <strong>ASPxClientUploadControlFileUploadCompleteEventHandler.callbackData</strong> received from server. </p>
+3. Add the [ASPxUploadControl](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxUploadControl?p=netframework) to the page. Set the control's [ShowUploadButton](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxUploadControl.ShowUploadButton?p=netframework) property to `true` to allow users to upload files. Use the [ValidationSettings.AllowedFileExtensions](https://docs.devexpress.com/AspNet/DevExpress.Web.UploadControlValidationSettings.AllowedFileExtensions?p=netframework) property to prevent users from uploading files other than images.
 
-<br/>
+    ```aspx
+    <dx:ASPxUploadControl ID="Upload" runat="server" OnFileUploadComplete="Upload_FileUploadComplete" 
+        ShowUploadButton="True" >
+        <ValidationSettings AllowedFileExtensions=".jpg,.jpeg,.jpe,.gif">
+        </ValidationSettings>
+        <ClientSideEvents FileUploadComplete="OnFileUploadComplete" />
+    </dx:ASPxUploadControl>
+    ```
 
+4. Handle the Upload Control's server-side [FileUploadComplete](https://docs.devexpress.com/AspNet/DevExpress.Web.ASPxUploadControl.FileUploadComplete?p=netframework) event. The [UploadedFile](https://docs.devexpress.com/AspNet/DevExpress.Web.FileUploadCompleteEventArgs.UploadedFile?p=netframework) event argument contains the uploaded image. Call the image's [SaveAs](https://docs.devexpress.com/AspNet/DevExpress.Web.UploadedFile.SaveAs(System.String)) method to save the image on the server. Set the [CallbackData](https://docs.devexpress.com/AspNet/DevExpress.Web.FileUploadCompleteEventArgs.CallbackData) property to the relative image path to pass it to the client.
 
+    ```csharp
+    protected void Upload_FileUploadComplete(object sender, FileUploadCompleteEventArgs e) {
+        // Assign a relative path to the 'CallbackData' property
+        // Relative path: "Images\Picture2023-02-17 10-19-58.jpg"
+        e.CallbackData = String.Format("Images\\Picture{0}.jpg", DateTime.Now.ToString("yyyy-MM-dd hh-mm-ss"));
+        
+        // Send a full path as the 'path' parameter to the 'SaveAs' method
```

Comment:
We would send something to the connected server, however, we can pass a full path to the `SaveAs` method...

Word order.

Again, I moved the aspect of circumstances to the end of the base sentence structure. Pass a full path to the method how? As the `path" parameter.

```suggestion
        // Pass a full path to the 'SaveAs' method as the 'path' parameter
```